### ISSUE TITLE
zipline ingest not working

### ISSUE THREAD

**Issue by plavine8183** [quantopian/zipline#2134]
Dear Zipline Maintainers,

Before I tell you about my issue, let me describe my environment:

# Environment

* Operating System: Linux  3.10.0-693.21.1.el7.x86_64 #1 SMP Wed Mar 7 19:03:37 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
* Python Version: 3.6.4
* Python Bitness: 64
* How did you install Zipline: pip
* Python packages: 

alembic==0.9.8
asn1crypto==0.24.0
bcolz==0.12.1
bcrypt==3.1.4
bleach==2.1.3
Bottleneck==1.2.1
certifi==2018.1.18
cffi==1.11.5
chardet==3.0.4
click==6.7
contextlib2==0.5.5
cryptography==2.1.4
cyordereddict==1.0.0
Cython==0.27.3
decorator==4.2.1
empyrical==0.4.1
entrypoints==0.2.3
flake8==3.5.0
html5lib==1.0.1
idna==2.6
inflection==0.3.1
intervaltree==2.1.0
ipykernel==4.8.2
ipython==6.2.1
ipython-genutils==0.2.0
ipywidgets==7.1.2
jedi==0.11.1
Jinja2==2.10
jsonschema==2.6.0
jupyter==1.0.0
jupyter-client==5.2.3
jupyter-console==5.2.0
jupyter-core==4.4.0
Logbook==1.3.0
lru-dict==1.1.6
lxml==4.1.1
Mako==1.0.7
MarkupSafe==1.0
mccabe==0.6.1
mistune==0.8.3
more-itertools==4.1.0
multipledispatch==0.5.0
nbconvert==5.3.1
nbformat==4.4.0
ndg-httpsclient==0.4.4
networkx==2.1
notebook==5.4.0
numexpr==2.6.4
numpy==1.14.2
pandas==0.18.1
pandas-datareader==0.6.0
pandocfilters==1.4.2
paramiko==2.4.0
parso==0.1.1
patsy==0.5.0
pexpect==4.4.0
pickleshare==0.7.4
prompt-toolkit==1.0.15
ptyprocess==0.5.2
pyasn1==0.4.2
pycodestyle==2.3.1
pycparser==2.18
pyflakes==1.6.0
Pygments==2.2.0
PyNaCl==1.2.1
pyOpenSSL==17.5.0
python-dateutil==2.7.2
python-editor==1.0.3
pytz==2018.3
PyYAML==3.12
pyzmq==17.0.0
qtconsole==4.3.1
Quandl==3.3.0
requests==2.18.4
requests-file==1.4.3
requests-ftp==0.3.1
scipy==1.0.0
Send2Trash==1.5.0
simplegeneric==0.8.1
six==1.11.0
sortedcontainers==1.5.9
SQLAlchemy==1.2.5
statsmodels==0.8.0
tables==3.4.2
terminado==0.8.1
testpath==0.3.1
toolz==0.9.0
tornado==5.0
traitlets==4.3.2
urllib3==1.22
wcwidth==0.1.7
webencodings==0.5.1
widgetsnbextension==3.1.4
wrapt==1.10.11
zipline==1.1.1

Now that you know a little about me, let me tell you about the issue I am
having:

# Description of Issue:

INGEST fails on pip installed version of zipline packaged with 18.1 pandas

[root@won-la1-net01 scripts]# QUANDL_API_KEY=XXXXX zipline ingest -b quandl
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 659, in _build_master
    ws.require(__requires__)
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 967, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 858, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (pandas 0.18.1 (/usr/local/lib/python3.6/site-packages), Requirement.parse('pandas>=0.19.2'), {'pandas-datareader'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/zipline", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3017, in <module>
    @_call_aside
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3003, in _call_aside
    f(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3030, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 661, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 674, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 858, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (pandas 0.18.1 (/usr/local/lib/python3.6/site-packages), Requirement.parse('pandas>=0.19.2'), {'pandas-datareader'})

Here is how you can reproduce this issue on your machine:

## Reproduction Steps

1. 
2.
3.
...

## What steps have you taken to resolve this already?

tried uninstalling pandas and used pip to upgrade it to 0.19.2

Sincerely,
Philip

**Comment by freddiev4** [comment]
@plavine8183 can you try making a Python 3.5 virtualenv? We don't have support for Python 3.6.

If you're not familiar with `virtualenv`, you can find the install guide [here](https://virtualenv.pypa.io/en/stable/installation/) along with the guide for `virtualenvwrapper` [here](http://virtualenvwrapper.readthedocs.io/en/latest/install.html).

If you have those installed successfully, and you have Python 3.5 on your machine, you can run

`mkvirtualenv zip35 -p /path/to/your/python3.5`

And then `pip install zipline`

**Comment by plavine8183** [comment]
new issue:

Downloading WIKI metadata:   [####################################]
Traceback (most recent call last):
  File "/opt/zip35/bin/zipline", line 11, in <module>
    sys.exit(main())
  File "/opt/zip35/lib/python3.5/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/opt/zip35/lib/python3.5/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/opt/zip35/lib/python3.5/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/zip35/lib/python3.5/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/zip35/lib/python3.5/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/opt/zip35/lib/python3.5/site-packages/zipline/__main__.py", line 312, in ingest
    show_progress,
  File "/opt/zip35/lib/python3.5/site-packages/zipline/data/bundles/core.py", line 451, in ingest
    pth.data_path([name, timestr], environ=environ),
  File "/opt/zip35/lib/python3.5/site-packages/zipline/data/bundles/quandl.py", line 295, in quandl_bundle
    show_progress=show_progress,
  File "/opt/zip35/lib/python3.5/site-packages/zipline/data/bundles/quandl.py", line 129, in fetch_symbol_metadata_frame
    data['auto_close_date'] = data['end_date'] + pd.Timedelta(days=1)
  File "/opt/zip35/lib/python3.5/site-packages/pandas/core/ops.py", line 609, in wrapper
    na_op)
  File "/opt/zip35/lib/python3.5/site-packages/pandas/core/ops.py", line 569, in maybe_convert_for_time_op
    return cls(left, right, name, na_op)
  File "/opt/zip35/lib/python3.5/site-packages/pandas/core/ops.py", line 282, in __init__
    lvalues = self._convert_to_array(left, name=name)
  File "/opt/zip35/lib/python3.5/site-packages/pandas/core/ops.py", line 398, in _convert_to_array
    if (inferred_type in ('datetime64', 'datetime', 'date', 'time') or
TypeError: data type "datetime" not understood

**Comment by freddiev4** [comment]
@plavine8183 what version of pandas do you have? You should have 0.18.1

**Comment by freddiev4** [comment]
Ah. Wait a minute I think we saw this issue before; this is fixed on the latest zipline master.

So, you can either wait for us to do a release (hoping to do it this week actually), or you can install the latest zipline master, assuming you have Git:

```
git clone https://github.com/quantopian/zipline.git
pip install zipline/
```

**Comment by plavine8183** [comment]
no workie - oh well I'll wait the week out

**Comment by freddiev4** [comment]
Can you provide more details? Did the install process not work? Or `ingest` after you installed the latest master?

**Comment by plavine8183** [comment]
the ingest did not work after the git cloned install

**Comment by freddiev4** [comment]
Can you show me the output when you run:
```
pip freeze
```

**Comment by plavine8183** [comment]
[root@won-la1-net01 bin]# git clone https://github.com/quantopian/zipline.git
fatal: destination path 'zipline' already exists and is not an empty directory.
[root@won-la1-net01 bin]# python3.5 -m pip install zipline
Collecting zipline
Collecting alembic>=0.7.7 (from zipline)
Collecting empyrical>=0.2.2 (from zipline)
  Using cached empyrical-0.4.2-py3-none-any.whl
Collecting multipledispatch>=0.4.8 (from zipline)
  Using cached multipledispatch-0.5.0-py2.py3-none-any.whl
Collecting pandas<0.19,>=0.18.1 (from zipline)
  Using cached pandas-0.18.1-cp35-cp35m-manylinux1_x86_64.whl
Collecting requests>=2.9.1 (from zipline)
  Using cached requests-2.18.4-py2.py3-none-any.whl
Collecting toolz>=0.8.2 (from zipline)
Collecting scipy>=0.17.1 (from zipline)
  Using cached scipy-1.0.1-cp35-cp35m-manylinux1_x86_64.whl
Collecting patsy>=0.4.0 (from zipline)
  Using cached patsy-0.5.0-py2.py3-none-any.whl
Collecting requests-file>=1.4.1 (from zipline)
  Using cached requests_file-1.4.3-py2.py3-none-any.whl
Collecting lru-dict>=1.1.4 (from zipline)
Collecting numexpr>=2.6.1 (from zipline)
  Using cached numexpr-2.6.4-cp35-cp35m-manylinux1_x86_64.whl
Collecting intervaltree>=2.1.0 (from zipline)
Collecting sqlalchemy>=1.0.8 (from zipline)
Requirement already satisfied: pip>=7.1.0 in /usr/local/lib/python3.5/site-packages (from zipline)
Collecting pytz>=2016.4 (from zipline)
  Using cached pytz-2018.3-py2.py3-none-any.whl
Collecting sortedcontainers>=1.4.4 (from zipline)
  Using cached sortedcontainers-1.5.9-py2.py3-none-any.whl
Collecting pandas-datareader>=0.2.1 (from zipline)
  Using cached pandas_datareader-0.6.0-py2.py3-none-any.whl
Collecting statsmodels>=0.6.1 (from zipline)
  Using cached statsmodels-0.8.0-cp35-cp35m-manylinux1_x86_64.whl
Collecting python-dateutil>=2.4.2 (from zipline)
  Using cached python_dateutil-2.7.2-py2.py3-none-any.whl
Collecting MarkupSafe>=0.23 (from zipline)
Collecting click>=4.0.0 (from zipline)
  Using cached click-6.7-py2.py3-none-any.whl
Collecting bcolz<1,>=0.12.1 (from zipline)
Requirement already satisfied: six>=1.10.0 in /usr/local/lib/python3.5/site-packages (from zipline)
Requirement already satisfied: setuptools>18.0 in /usr/local/lib/python3.5/site-packages (from zipline)
Collecting contextlib2>=0.4.0 (from zipline)
  Using cached contextlib2-0.5.5-py2.py3-none-any.whl
Collecting numpy>=1.11.1 (from zipline)
  Using cached numpy-1.14.2-cp35-cp35m-manylinux1_x86_64.whl
Collecting cyordereddict>=0.2.2 (from zipline)
Collecting tables>=3.3.0 (from zipline)
  Using cached tables-3.4.2-3-cp35-cp35m-manylinux1_x86_64.whl
Collecting bottleneck>=1.0.0 (from zipline)
Collecting networkx>=1.9.1 (from zipline)
Collecting Logbook>=0.12.5 (from zipline)
Collecting Cython>=0.25.2 (from zipline)
  Using cached Cython-0.28.1-cp35-cp35m-manylinux1_x86_64.whl
Collecting decorator>=4.0.0 (from zipline)
  Using cached decorator-4.2.1-py2.py3-none-any.whl
Collecting Mako>=1.0.1 (from zipline)
Collecting python-editor>=0.3 (from alembic>=0.7.7->zipline)
Collecting urllib3<1.23,>=1.21.1 (from requests>=2.9.1->zipline)
  Using cached urllib3-1.22-py2.py3-none-any.whl
Collecting chardet<3.1.0,>=3.0.2 (from requests>=2.9.1->zipline)
  Using cached chardet-3.0.4-py2.py3-none-any.whl
Collecting idna<2.7,>=2.5 (from requests>=2.9.1->zipline)
  Using cached idna-2.6-py2.py3-none-any.whl
Collecting certifi>=2017.4.17 (from requests>=2.9.1->zipline)
  Using cached certifi-2018.1.18-py2.py3-none-any.whl
Collecting requests-ftp (from pandas-datareader>=0.2.1->zipline)
Collecting lxml (from pandas-datareader>=0.2.1->zipline)
  Using cached lxml-4.2.1-cp35-cp35m-manylinux1_x86_64.whl
Collecting wrapt (from pandas-datareader>=0.2.1->zipline)
Installing collected packages: MarkupSafe, Mako, python-editor, sqlalchemy, python-dateutil, alembic, urllib3, chardet, idna, certifi, requests, requests-ftp, lxml, wrapt, numpy, pytz, pandas, requests-file, pandas-datareader, bottleneck, scipy, empyrical, multipledispatch, toolz, patsy, lru-dict, numexpr, sortedcontainers, intervaltree, statsmodels, click, bcolz, contextlib2, cyordereddict, tables, decorator, networkx, Logbook, Cython, zipline
Successfully installed Cython-0.28.1 Logbook-1.3.0 Mako-1.0.7 MarkupSafe-1.0 alembic-0.9.9 bcolz-0.12.1 bottleneck-1.2.1 certifi-2018.1.18 chardet-3.0.4 click-6.7 contextlib2-0.5.5 cyordereddict-1.0.0 decorator-4.2.1 empyrical-0.4.2 idna-2.6 intervaltree-2.1.0 lru-dict-1.1.6 lxml-4.2.1 multipledispatch-0.5.0 networkx-2.1 numexpr-2.6.4 numpy-1.14.2 pandas-0.18.1 pandas-datareader-0.6.0 patsy-0.5.0 python-dateutil-2.7.2 python-editor-1.0.3 pytz-2018.3 requests-2.18.4 requests-file-1.4.3 requests-ftp-0.3.1 scipy-1.0.1 sortedcontainers-1.5.9 sqlalchemy-1.2.6 statsmodels-0.8.0 tables-3.4.2 toolz-0.9.0 urllib3-1.22 wrapt-1.10.11 zipline-1.1.1
You are using pip version 9.0.1, however version 9.0.3 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
[root@won-la1-net01 bin]# source activate
(zip35) [root@won-la1-net01 bin]# QUANDL_API_KEY=25WXi4prssqfy1vzUKxN zipline ingest -b quandl
Downloading WIKI metadata:   [####################################]
Traceback (most recent call last):
  File "/usr/local/bin/zipline", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.5/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.5/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.5/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.5/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.5/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.5/site-packages/zipline/__main__.py", line 312, in ingest
    show_progress,
  File "/usr/local/lib/python3.5/site-packages/zipline/data/bundles/core.py", line 451, in ingest
    pth.data_path([name, timestr], environ=environ),
  File "/usr/local/lib/python3.5/site-packages/zipline/data/bundles/quandl.py", line 295, in quandl_bundle
    show_progress=show_progress,
  File "/usr/local/lib/python3.5/site-packages/zipline/data/bundles/quandl.py", line 129, in fetch_symbol_metadata_frame
    data['auto_close_date'] = data['end_date'] + pd.Timedelta(days=1)
  File "/usr/local/lib/python3.5/site-packages/pandas/core/ops.py", line 609, in wrapper
    na_op)
  File "/usr/local/lib/python3.5/site-packages/pandas/core/ops.py", line 569, in maybe_convert_for_time_op
    return cls(left, right, name, na_op)
  File "/usr/local/lib/python3.5/site-packages/pandas/core/ops.py", line 282, in __init__
    lvalues = self._convert_to_array(left, name=name)
  File "/usr/local/lib/python3.5/site-packages/pandas/core/ops.py", line 398, in _convert_to_array
    if (inferred_type in ('datetime64', 'datetime', 'date', 'time') or
TypeError: data type "datetime" not understood

**Comment by plavine8183** [comment]
[root@won-la1-net01 bin]# python3.5 -m pip freeze
alembic==0.9.9
bcolz==0.12.1
Bottleneck==1.2.1
certifi==2018.1.18
chardet==3.0.4
click==6.7
contextlib2==0.5.5
cyordereddict==1.0.0
Cython==0.28.1
decorator==4.2.1
empyrical==0.4.2
idna==2.6
intervaltree==2.1.0
Logbook==1.3.0
lru-dict==1.1.6
lxml==4.2.1
Mako==1.0.7
MarkupSafe==1.0
multipledispatch==0.5.0
networkx==2.1
numexpr==2.6.4
numpy==1.14.2
pandas==0.18.1
pandas-datareader==0.6.0
patsy==0.5.0
pbr==4.0.0
python-dateutil==2.7.2
python-editor==1.0.3
pytz==2018.3
requests==2.18.4
requests-file==1.4.3
requests-ftp==0.3.1
scipy==1.0.1
six==1.11.0
sortedcontainers==1.5.9
SQLAlchemy==1.2.6
statsmodels==0.8.0
stevedore==1.28.0
tables==3.4.2
toolz==0.9.0
urllib3==1.22
virtualenv==15.2.0
virtualenv-clone==0.3.0
virtualenvwrapper==4.8.2
wrapt==1.10.11
zipline==1.1.1
You are using pip version 9.0.1, however version 9.0.3 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.

**Comment by freddiev4** [comment]
Ah. I see where the issue is. You ran:
```
python3.5 -m pip install zipline
```

Which still installs zipline from PyPi.

You need to run
```
python3.5 -m pip install zipline/
```

The forward slash indicates that you're installing your local clone of the zipline repository.

**Comment by plavine8183** [comment]
zip35) [root@won-la1-net01 bin]# QUANDL_API_KEY=XXXX zipline ingest -b quandl
[2018-04-02 19:53:26.982953] INFO: zipline.data.bundles.quandl: Downloading WIKI metadata.
Downloading WIKI Prices table from Quandl  [####################################]  100%             
[2018-04-02 19:54:37.608147] INFO: zipline.data.bundles.quandl: Parsing raw data.
[2018-04-02 19:55:21.663488] INFO: zipline.data.bundles.quandl: Generating asset metadata.
Merging daily equity files:  [--------#---------------------------]  1731/usr/local/lib/python3.5/site-packages/zipline/data/us_equity_pricing.py:417: UserWarning: Ignoring 1 values because they are out of bounds for uint32:             open  high   low  close        volume  ex_dividend  split_ratio
2011-04-11  1.79  1.84  1.55    1.7  6.674913e+09          0.0          1.0
  winsorise_uint32(raw_data, invalid_data_behavior, 'volume', *OHLC)
Merging daily equity files:  [####################################]      
[2018-04-02 20:00:33.068948] INFO: zipline.data.bundles.quandl: Parsing split data.
[2018-04-02 20:00:33.438012] INFO: zipline.data.bundles.quandl: Parsing dividend data.
/usr/local/lib/python3.5/site-packages/zipline/data/us_equity_pricing.py:931: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  if not issubdtype(actual, expected):
(zip35) [root@won-la1-net01 bin]# 


I guess this is a little better?

**Comment by freddiev4** [comment]
Going to close this as it looks like this worked.

We also recently did a release of zipline. You can see the release notes [here](http://www.zipline.io/releases.html#release-1-2-0) Feel free to update to 1.2.0 with either:

```
pip install -U zipline
```

or

```
conda update zipline -c quantopian
```

If there are any problems, please open a new issue 🙂

**Comment by plavine8183** [comment]
Thank you - it is working

Sent from lunchbox.

On Apr 5, 2018, at 7:46 AM, Freddie Vargus <notifications@github.com<mailto:notifications@github.com>> wrote:


Closed #2134<https://github.com/quantopian/zipline/issues/2134>.
